### DISCUSSION TITLE
why prisma try to recreating the table when i delete single nullable column?

### DISCUSSION THREAD

**Discussion by tsulatsitamim** [prisma/prisma#24790] [Q&A]
### Question

why prisma try to recreating the table when i delete single nullable column? I use mysql. I try to delete summary column. there is no data in Survey table

```prisma
model Survey {
  id              Int              @id @default(autoincrement())
  externalId      String           @unique() @db.VarChar(21)
  title           String           @default("Untitled")
  description     String           @default("") @db.Text
  summary         Json?
  status          Status           @default(INACTIVE)
  type            String?
  departmentId    Int
  createdAt       DateTime         @default(now())
  updatedAt       DateTime         @updatedAt
  department      Department       @relation(fields: [departmentId], references: [id])
  surveyQuestions SurveyQuestion[]
  surveyResponses SurveyResponse[]
}
```

**Comment by ludralph** [comment]
Hi @tsulatsitamim 👋 

Thank you for raising this question.

When you delete a column from a table in Prisma Schema, Prisma Schema will recreate the table to ensure the schema changes are applied correctly. This behavior can be observed even if the column is nullable and there is no data in the table. You are trying to delete the `summary` column from the `Survey` model. Prisma's migration engine will recreate the table to apply this change. This is a common approach to ensure data integrity and consistency, especially when dealing with relational databases like MySQL. YOu can learn more about this from the [documentation](https://www.prisma.io/docs/orm/prisma-migrate/workflows/data-migration)

**Reply by tsulatsitamim** [reply]
If this is normal then I run migrate deploy, my production data will gone

**Reply by ludralph** [reply]
Hi @tsulatsitamim 

If you run `prisma migrate deploy` in a production environment, it should not reset your database or cause data loss. The `prisma migrate deploy` command is designed to apply migrations without resetting the database, which is crucial for maintaining data integrity in production environments. 

`migrate dev` is about creating new migrations and making sure your development database is in sync. We assume the data on the development database does not matter, so `migrate dev` will ask you to reset the database whenever necessary (if you changed a migration that was already applied, or if you switched branches and suddenly migrations "disappeared").

For other databases, that of course should never be reset in the normal workflow, you should be using `prisma migrate deploy`. This will never reset, it just applies migrations.

**Reply by tsulatsitamim** [reply]
prisma migrate dev create sql file that include recreating my table, if i run prisma migrate deploy, it will run that sql file which will recreate my production table and loss the data

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
